### ISSUE TITLE
Bug: The property ignore seems to be missing in the type ConfigWithExtends used in the config api

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#8562]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

```TypeScript
The property `ignore` seems to be missing in the type definition of `ConfigWithExtends` used in the `config` when using the eslint flat config format api.


// @ts-check

import tseslint from 'typescript-eslint';

export default tseslint.config({
	ignore: [
		'vendor/**/*',
	],
});
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

n.a.

### Actual Result

n.a.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `7.1.0` |
| `@typescript-eslint/parser`            | `7.1.0` |
| `TypeScript`                           | `5.3.3` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `20.11.0` |

**Comment by doberkofler** [comment]
Sorry! My mistake. i totally overlooked that i wrote `ignore` instead of `ignores`!
